qa/tasks/cephfs/mount.py

```diff
@@ -234,9 +234,9 @@ def open_background(self, basename="background_file", write=True):
                 import time
 
                 f = open("{path}", 'w')
-                f.write('content')
+                f.write(b'content')
```

Comment:
if `f` is opened in text mode, i.e. it's not opened in binary mode like
```py
f = open("{path}", 'wb')
```
i think the existing code is correct as it is.

the same applies to other places where you are changing the code to
```py
f.write(an_instance_of_bytes)
```

Reply:
Huh, I was confused by this failure: /ceph/teuthology-archive/pdonnell-2019-12-20_23:53:25-fs-wip-pdonnell-testing-20191220.212030-distro-basic-smithi/4620237/teuthology.log

````
2019-12-21T01:33:22.376 INFO:teuthology.orchestra.run.smithi089:> sudo adjust-ulimits daemon-helper kill python3 -c '
2019-12-21T01:33:22.376 INFO:teuthology.orchestra.run.smithi089:> import zlib
2019-12-21T01:33:22.376 INFO:teuthology.orchestra.run.smithi089:> path = "/home/ubuntu/cephtest/mnt.0/subdir/0"
2019-12-21T01:33:22.377 INFO:teuthology.orchestra.run.smithi089:> f = open(path, '"'"'w'"'"')
2019-12-21T01:33:22.377 INFO:teuthology.orchestra.run.smithi089:> for i in range(0, 6291456):
2019-12-21T01:33:22.377 INFO:teuthology.orchestra.run.smithi089:>     val = zlib.crc32("%s" % i) & 7
2019-12-21T01:33:22.377 INFO:teuthology.orchestra.run.smithi089:>     f.write(chr(val))
2019-12-21T01:33:22.377 INFO:teuthology.orchestra.run.smithi089:> f.close()
2019-12-21T01:33:22.377 INFO:teuthology.orchestra.run.smithi089:> '
2019-12-21T01:33:22.495 INFO:teuthology.orchestra.run.smithi089.stderr:Traceback (most recent call last):
2019-12-21T01:33:22.495 INFO:teuthology.orchestra.run.smithi089.stderr:  File "<string>", line 6, in <module>
2019-12-21T01:33:22.496 INFO:teuthology.orchestra.run.smithi089.stderr:TypeError: a bytes-like object is required, not 'str'
2019-12-21T01:33:22.676 INFO:teuthology.orchestra.run.smithi089.stderr:daemon-helper: command failed with exit status 1
````

I see now I just had the line numbers wrong. The erroneous argument is in the call to `zlib.crc32`.

Reply:
Fixed!